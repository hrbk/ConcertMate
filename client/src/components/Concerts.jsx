import React from 'react';
import axios from 'axios';
import ConcertEntry from './ConcertEntry.jsx';
import {ListGroup} from 'react-bootstrap';
import {Panel, Tabs, Tab} from 'react-bootstrap';
import { connect } from 'react-redux';
import ReactScrollbar from 'react-scrollbar-js';

const Concerts = (props) => {
  const scroll = {
    width: '100%',
    height: '400',
    speed: '10'
  }
  const nearby = (
    <ReactScrollbar style={scroll}>
      <Panel collapsible defaultExpanded header={`Nearby Concerts`}>
        <ListGroup fill>
          {props.events.map((event, i) => {
            return <ConcertEntry event={event} key={i} handleArtistClick={props.handleArtistClick} />
          })}
        </ListGroup>
      </Panel>
    </ReactScrollbar>
  )
  const popularityObj = (events) => {
    props.events.sort((a, b) => {
      if (a.popularity > b.popularity) {
        return 1;
      }
      if (a.popularity < b.popularity) {
        return -1;
      }
      return 0;
    }).reverse();
  }

  const popular = (
    <ReactScrollbar style={scroll}>
      <Panel collapsible defaultExpanded header={`Popular Concerts`}>
        <ListGroup fill>
          {popularityObj(props.events)}
          {props.events.map((event, i) => {
            return <ConcertEntry event={event} key={i} handleArtistClick={props.handleArtistClick} />
          })}
        </ListGroup>
      </Panel>
    </ReactScrollbar>
  )

  return (
    <div>
      <Tabs defaultActiveKey={1}>
        <Tab eventKey={1} title='Nearby'>{nearby}</Tab>
        <Tab eventKey={2} title='Popular'>{popular}</Tab>
      </Tabs>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    'events': state.events
  }
}

export const UnwrappedConcerts = Concerts;
export default connect(mapStateToProps)(Concerts);
