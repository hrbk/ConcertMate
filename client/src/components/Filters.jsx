import React from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import {Navbar, FormGroup, FormControl, Button} from 'react-bootstrap';
import PlacesAutocomplete, { geocodeByAddress, getLatLng } from 'react-places-autocomplete';
import axios from 'axios';
import { setDate, setSearchTerm, setMapCenter } from '../redux/actionCreators.js';
import { connect } from 'react-redux';


// ignore the fact that this is called Favorites but the file is called Filters
const Filters = (props) => {
  // beginning of search functionality. we wanted to implement google search to be able to
  // autocomplete addresses but hey that's your job now
  const handleSubmit = (loc) => {

    geocodeByAddress(loc)
    .then(results => getLatLng(results[0]))
    .then(latLng => {

      let userLoc = {
        lat: latLng.lat,
        lng: latLng.lng
      }
      props.handleSetMapCenter(userLoc); //set state of mapCenter prop
      props.requestEvents(); //getSongkickEvents

    })
    .catch(error => console.log('error', error))
  }

  const datepicker =  {
    paddingTop: '3.5px'
  }

  const inputProps = {
      value: props.searchTerm,
      onChange: props.handleSetSearchTerm,
      class: 'search_input',
      placeholder: 'Location'
  }

  const navStyle = {
    display: 'flex',
    alignItems: 'center'
  }

  return (
    <div className="navigation">
      <Navbar.Form pullRight style={navStyle}>
        <PlacesAutocomplete
          onEnterKeyDown={handleSubmit}
          inputProps={inputProps}
          highlightFirstSuggestion={true}
          googleLogo={false}/>

        <div style={datepicker}>
          <DatePicker
            dateFormat="MM/DD/YYYY"
            selected={props.date} //automatically passed into handleSetData
            onChange={props.handleSetDate} />
        </div>
        <Button type="submit" onClick={() => handleSubmit(props.searchTerm)}>Submit</Button>
      </Navbar.Form>
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    'date': state.date, //allows this.props.date to exist and be accessible in component
    'searchTerm': state.searchTerm,
    'mapCenter': state.mapCenter
  }
}

const mapDispatchToProps = (dispatch) => ({
  handleSetDate(date = moment()) { //think of this as used in place of 'this.setState()'
    dispatch(setDate(date)); //action creator dispatched and reducers called: { type: 'SET_SPOTIFY_TOKEN', value: token } equivalent
  },
  handleSetSearchTerm(searchTerm = '') {
    dispatch(setSearchTerm(searchTerm)); //no need for searchTerm.target.value bc googleAutoPlaces handles that
  },
  handleSetMapCenter(mapCenter) {
    dispatch(setMapCenter(mapCenter));
  }
});

export const UnwrapperFilters = Filters;
export default connect(mapStateToProps, mapDispatchToProps)(Filters);
