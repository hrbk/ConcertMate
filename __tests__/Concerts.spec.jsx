import React from 'react';
import renderer from 'react-test-renderer';
import Filters from '../client/src/components/Filters.jsx';
import DatePicker from 'react-datepicker';
import { shallow, mount } from 'enzyme';
import Concerts, { UnwrappedConcerts } from '../client/src/components/Concerts.jsx';
import ConcertEntrys, { UnWrappedConcertEntry } from '../client/src/components/ConcertEntry.jsx';
import {ListGroup} from 'react-bootstrap';

describe('<Concerts />', () => {
  it('should render correctly', () => {
    const mockedEvents = [];
    const component = renderer.create(<UnwrappedConcerts events={mockedEvents} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render 10 ConcertEntrys given 10 events', () => {
    const mockedEvent = {"ageRestriction":"All Ages","displayName":"Roy Gaines at Biscuits and Blues (August 26, 2017)","id":30615579,"popularity":0.00014,"uri":"http://www.songkick.com/concerts/30615579-roy-gaines-at-biscuits-and-blues?utm_source=46052&utm_medium=partner"};
    const mockedEvents = [];
    for (var i = 0; i < 10; i++) {
      mockedEvents.push(mockedEvent);
    }
    const component = shallow(<UnwrappedConcerts events={mockedEvents} />);
    expect(component.find(ConcertEntrys).length).toEqual(10);
  });
})
