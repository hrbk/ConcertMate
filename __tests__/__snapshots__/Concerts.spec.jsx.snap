// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Concerts /> should render correctly 1`] = `
<div
  className="panel panel-default"
  id={null}
>
  <div
    className="panel-heading"
  >
    <h4
      className="panel-title"
      role="presentation"
    >
      <a
        aria-controls={undefined}
        aria-expanded={true}
        aria-selected={true}
        className={null}
        href={undefined}
        onClick={[Function]}
        role={undefined}
      >
        Nearby Concerts
      </a>
    </h4>
  </div>
  <div
    aria-expanded={null}
    aria-hidden={false}
    className="panel-collapse collapse in"
    id={undefined}
    role={undefined}
  >
    <ul
      className="list-group"
      fill={undefined}
    />
  </div>
</div>
`;
